src/node/interfaces.cpp

```diff
@@ -536,6 +537,20 @@ class ChainImpl : public Chain
         }
         return std::nullopt;
     }
+    bool hasBlockFilterIndex(BlockFilterType filter_type) override
+    {
+        return GetBlockFilterIndex(filter_type) != nullptr;
+    }
+    std::optional<bool> blockFilterMatchesAny(BlockFilterType filter_type, const uint256& block_hash, const GCSFilter::ElementSet& filter_set) override
+    {
+        const BlockFilterIndex* block_filter_index = GetBlockFilterIndex(filter_type);
+        if (!block_filter_index) return std::nullopt;
+
+        BlockFilter filter;
+        const CBlockIndex* index = WITH_LOCK(::cs_main, return chainman().m_blockman.LookupBlockIndex(block_hash));
+        if (!block_filter_index->LookupFilter(index, filter)) return std::nullopt;
```

Comment:
No need to add `block_filter_index->BlockUntilSyncedToCurrentChain` ?
Currently it seems to assume the chain is already synced.

```suggestion
        if (!block_filter_index->BlockUntilSyncedToCurrentChain() || !block_filter_index->LookupFilter(index, filter)) return std::nullopt;
```

Reply:
Reviewing the code again, it seems to me that if the index not fully synced, the wallet will normally scan the remaining blocks. In this case, this change is really not necessary.